addons/binding/org.openhab.binding.hp1000/src/main/java/org/openhab/binding/hp1000/handler/StateConverterUtils.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.hp1000.handler;
+
+import static org.openhab.binding.hp1000.HP1000BindingConstants.WEBHOOK_DATE_TIME_FORMAT;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+
+/**
+ * The {@link StateConverterUtils} is a helper to convert values to state
+ *
+ * @author Daniel Bauer - Initial contribution
+ */
+final class StateConverterUtils {
+
+    @Nullable
+    public static Double parseDouble(String value) {
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+
+        return Double.valueOf(value);
+    }
+
+    public static State toDateTimeType(@Nullable String value) {
+        DateTimeType dateTimeType;
+        try {
+            dateTimeType = new DateTimeType(value);
+        } catch (IllegalArgumentException exception) {
+            DateTimeFormatter parser = DateTimeFormatter.ofPattern(WEBHOOK_DATE_TIME_FORMAT)
+                    .withZone(ZoneId.systemDefault());
```

Comment:
I am not sure if it is a good idea to just pick the default timezone. Maybe you want to inject the `TimeZoneProvider` into your handlerfactory (since this is an osgi service) and then hand it over to your handler via a constructor parameter. You can then query that one for the correct timezone.